external/model-preparation-algorithm/mpa_tasks/apis/task.py

```diff
@@ -155,10 +156,48 @@ def _initialize(self, dataset=None, output_model=None, export=False):
         """
         logger.info('initializing....')
         self._init_recipe()
+
         if not export:
             recipe_hparams = self._init_recipe_hparam()
             if len(recipe_hparams) > 0:
                 self._recipe_cfg.merge_from_dict(recipe_hparams)
+
+            # Put early stop hook
+            if self._recipe_cfg.early_stop is True:
+                patience = self._recipe_cfg.patience
+                metric = self._recipe_cfg.early_stop_metric
+                
+                custom_hooks = self._recipe_cfg.get('custom_hooks',[])
+                early_stop_hook = dict(
+                            type='LazyEarlyStoppingHook',
+                            start=3,
+                            patience=patience,
+                            iteration_patience=0,
+                            interval=1,
+                            metric=metric,
+                            priority=75,
+                        )
+
+                for hook in custom_hooks:
+                    if hook.type in ['EarlyStoppingHook','LazyEarlyStoppingHook']:
+                        early_stop_hook = hook
+                        break
+                
+                early_stop_hook['patience'] = patience
+                early_stop_hook['metric'] = metric
+
+                self.update_override_configurations({"custom_hooks" : [early_stop_hook]})
+            else:
+                custom_hooks = self._recipe_cfg.get('custom_hooks',[])
```

Comment:
I think this can be solved simply by increasing the start value of early-stop to more than max_epoch. What do you think?

Reply:
Thank you for suggestion, but it kind of looks like a hack. Completely removing the early stopping hook can also prevent from redundant part such as compare between current epoch and start epoch for every epoch in early stopping hook.